ls-ui-react-app/src/appState.ts

```diff
@@ -8,6 +8,8 @@ class AppState {
     compilationResult? : CompilationResult
     errorMsg? : string // An optional error message to show to the user
     compilationIsLoading : boolean = false
+    request? : boolean = false
+    apiResponse : any
```

Comment:
See my comment on `apiResponse` below. `any` is generally considered an anti-pattern.